torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -0,0 +1,411 @@
+            "If this would be helpful for you, please open an issue at https://github.com/pytorch/vision/issues."
+        )
+
+    image, target = sample
+    return image, datapoints.Mask(F.to_image_tensor(target).squeeze(0))
+
+
+_WRAPPERS[datasets.SBDataset] = sbd_wrapper
+
+
+def celeba_wrapper(dataset, sample):
+    image, target = sample
+    return wrap_target_by_type(
+        dataset,
+        target,
+        {
+            "identity": datapoints.Label,
+            "bbox": lambda item: datapoints.BoundingBox(
+                item, format=datapoints.BoundingBoxFormat.XYWH, spatial_size=(image.height, image.width)
+            ),
+        },
+        # FIXME: Failing on "attr" here is problematic, since it is the default
+        fail_on=["attr", "landmarks"],
+    )
+
+
+_WRAPPERS[datasets.CelebA] = celeba_wrapper
+
+KITTI_CATEGORIES = ["Car", "Van", "Truck", "Pedestrian", "Person_sitting", "Cyclist", "Tram", "Misc", "DontCare"]
+KITTI_CATEGORY_TO_IDX = dict(zip(KITTI_CATEGORIES, range(len(KITTI_CATEGORIES))))
+
+
+def kitti_wrapper(dataset, sample):
+    image, target = sample
+
+    target = list_of_dicts_to_dict_of_lists(target)
+
+    target["boxes"] = datapoints.BoundingBox(
+        target["bbox"], format=datapoints.BoundingBoxFormat.XYXY, spatial_size=(image.height, image.width)
+    )
+    target["labels"] = datapoints.Label(
+        [KITTI_CATEGORY_TO_IDX[category] for category in target["type"]], categories=KITTI_CATEGORIES
+    )
+
+    return image, target
+
+
+_WRAPPERS[datasets.Kitti] = kitti_wrapper
+
+
+def oxford_iiit_pet_wrapper(dataset, sample):
+    image, target = sample
+    return image, wrap_target_by_type(
+        dataset,
+        target,
+        {
+            "category": lambda item: datapoints.Label(item, categories=dataset.classes),
+            "segmentation": lambda item: datapoints.Mask(F.pil_to_tensor(item).squeeze(0)),
+        },
+        attr_name="_target_types",
+    )
+
+
+_WRAPPERS[datasets.OxfordIIITPet] = oxford_iiit_pet_wrapper
+
+
+def cityscapes_wrapper(dataset, sample):
+    def instance_segmentation_wrapper(mask):
+        # See https://github.com/mcordts/cityscapesScripts/blob/8da5dd00c9069058ccc134654116aac52d4f6fa2/cityscapesscripts/preparation/json2instanceImg.py#L7-L21
+        data = F.pil_to_tensor(mask).squeeze(0)
+        masks = []
+        labels = []
+        for id in data.unique():
+            masks.append(data == id)
+            label = id
+            if label >= 1_000:
+                label //= 1_000
+            labels.append(label)
+        masks = datapoints.Mask(torch.stack(masks))
+        # FIXME: without the labels, returning just the instance masks is pretty useless. However, we would need to
+        #  return a two-tuple or the like where we originally only had a single PIL image.
+        labels = datapoints.Label(torch.stack(labels), categories=[cls.name for cls in dataset.classes])
```

Comment:
I'm not sure how the instance segmentations are used in the wild, so this would be a prime candidate to fail loudly and ask for user feedback. However, it is the default

https://github.com/pytorch/vision/blob/135a0f9ea9841b6324b4fe8974e2543cbb95709a/torchvision/datasets/cityscapes.py#L109

and thus failing here would make the dataset in its default form unusable with the wrapper.